starter-code/src/functions-and-arrays.js

```diff
@@ -10,14 +13,43 @@ var words = [
   'orchard',
   'crackpot'
 ];
+function findLongestWord (array){
+  var aux = '';
+  for(i = 0; i < array.length; i++){
+    if(array[i].length > aux.length){
+      aux = array[i];
+    }
+  }
+  if(array.length === 0){
+    return undefined;
+  }
+  return aux;
+}
 
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray (array){
+  var sum = 0;
+    for (i = 0; i < array.length; i++){
+    sum += array[i];
+    }
+      return sum;
+  }
 // Calculate the Average
 
 var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+function averageNumbers (array){
+  var sum = 0;
+    for (i = 0; i < array.length; i++){
+    sum += array[i];
+    }
+    if (array.length === 0){
+      return undefined;
+    }
+      return sum/array.length;
+  }
```

Comment:
Muy bien, pero ya sabéis, la evaluación mejor al principio y podéis usar reduce( ) para acortar líneas